### ISSUE TITLE
Browser: Close Tab Functionality

### ISSUE THREAD

**Issue by 2075** [cookiengineer/stealth-browser#23]
User wants to close a tab in the sidebar.

Context Menu or a little X on the side when hovering the element would do.

**Comment by cookiengineer** [comment]
Indeed it would be nice to have a better learning experience when it comes to the usage of shortcuts and mouse/touch interactions. Maybe something like a `[?]` icon in the bottom of the sidebar could trigger the help overlay?

Tabs currently can be closed by double-clicking or double-touching them or by pressing `[F6]`.

**Comment by cookiengineer** [comment]
As of https://github.com/cookiengineer/stealth/commit/ba4a48d7ed4ec043483525225005c1883affb676 the context menu has a Close Tab functionality, which integrates the `browser.close()` action with others on right click.

What's missing for mobile/tablet usage is a long-touch integration to open the context menu as there's no right click possible.

Safari on iOS specifically is broken here, the upstream bug is:

- https://bugs.webkit.org/show_bug.cgi?id=213953

**Comment by cookiengineer** [comment]
Upstream comment from WebKit developer(s):

```
AX: contextmenu event fired on Mac (w/VoiceOver) but not iOS (w/VoiceOver)

On Mac with VO, pressing VO+Shift+M (show contextual menu) triggers the `contextmenu` JavaScript event, equivalent to a mouse-triggered right-click or Ctrl+click. I assume this is because of a Mac implementation detail, and that it's been this way since `contextmenu` was implemented in WebKit... This is expected behavior on Mac.

On iOS, there are accessibility-triggered ways to show context menus in native UI: Home Screen icons for example. VoiceOver's default gesture for this is single-finger triple-tap. It used to be long-press, as mentioned by "cookiengineer" (no relation). The iOS VO key press is the same as Mac: VO+Shift+M. Yet neither of these trigger in Safari or other WebKit views.

I don't think this is purely an accessibility issue. As far as I am aware, there is no mainstream UI way to trigger a `contextmenu` event in iOS WebKit. As such, we should not add an accessibility-only way to trigger this event, as it could reveal the use of assistive technology.

Perhaps this bug should be blocked on a new WebKit issue to support `contextmenu` with a mainstream gesture or keypress if appropriate. (Disclaimer: I have not fully considered the implications, so there may be unexpected downsides.)
```

**Comment by cookiengineer** [comment]
I'm closing this issue as it depends on upstream Safari/WebKit supporting the `contextmenu` event, and there's nothing I can do to fix this without that integrated functionality except to simply not use Safari.

I won't start with custom polyfills for UX events, because that's not what users expect; and will break more than it will solve down the line.
